libtelemetry/src/main/java/com/mapbox/services/android/telemetry/MapboxTelemetry.java

```diff
@@ -196,9 +200,11 @@ private boolean startTelemetry() {
 
   private boolean optLocationIn() {
     if (isTelemetryEnabled && !isOpted) {
-      startLocation();
       registerEventReceiver();
       isOpted = true;
+      if (checkLocationPermission()) {
```

Comment:
Adding the check here would mean that
https://github.com/mapbox/mapbox-events-android/blob/22f1465aebbc21bb4924a6cfbf8a3879e1592754/libtelemetry/src/main/java/com/mapbox/services/android/telemetry/MapboxTelemetry.java#L203-L204 will be executed and we don't want that.

No need to register the receiver in charge of dealing with the location probes if permissions are not granted. Besides that, `isOpted` should only be `true` if previous actions `startLocation()` and `registerEventReceiver()` were executed, meaning that location is opted. If not, we may have issues when `optLocationOut()`.